### ISSUE TITLE
Idea: launcher GUI

### ISSUE THREAD

**Issue by markkurikola** [Interrupt/systemshock#118]
3RD and I agreed on Discord that a launcher GUI that allows the player to choose various parameters on startup would be a great addition. Examples: 

- OpenGL on/off
    - texture filtering on/off
    - other possible OpenGL specific features on/off
- fullscreen vs. windowed
- resolution selector
- show or hide console output
- mod selector
- save file to continue from, possibly picking up all of the above from that save's preferences

Making such a thing is of course time consuming, but would go a long way toward user friendliness.

**Comment by winterheart** [comment]
There no need external launcher. We may extend options menu with our configuration.

**Comment by Interrupt** [comment]
I agree that an external launcher isn't required for things that we could add to the in game options menu. A little window that would let people point their game at the data files without requiring someone to copy the files in to the right folder would be nice though, especially on platforms like OSX where the app is going to be packaged up into a .app folder that isn't straightforward to edit.

**Comment by mrikola** [comment]
Yeah, the entire point here was to suggest a convenience utility for things that would otherwise require command line parameters (mods), tweaking the internals of the game itself (eg. resolutions), or that would be cumbersome to change after the game has already started (eg. console output).

I kind of believe that this would be more convenient to develop, too, than modifying the in-game UI to fiddle with the aforementioned values. I'm honestly kind of surprised at the negative reaction.

**Comment by JPLeBreton** [comment]
One project to look at for "create a simple OS-independent window to prompt the user for a simple question on startup" would be GZDoom. It does introduce a UI toolkit into the dependencies though, which could add major headaches to the build process.

**Comment by AcceptableSpare** [comment]
Having to configure a game outside of it is very annoying especially when you want to test your controls out. I don't think it would ruin the ascetic of the game if you changed the escape menu to actually be useful and have stuff like configuring controls and more video options. You could move the pause logo up a bit and use a list instead of those boxes. Heck maybe have a toggle to use the old menu if people want it to look like the original game

**Comment by HunterZ** [comment]
Compromise: Maybe just switch to a plaintext config file format, with all options documented?
